# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/hyprpaper/hotfix-libjpeg.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- hyprpaper-0.7.1/src/helpers/Jpeg.cpp.vanilla	2024-11-22 23:05:57.276666633 +0100
+++ hyprpaper-0.7.1/src/helpers/Jpeg.cpp	2024-11-22 23:07:27.139999964 +0100
@@ -39,12 +39,12 @@
     decompressStruct.err = jpeg_std_error(&errorManager);
     jpeg_create_decompress(&decompressStruct);
     jpeg_mem_src(&decompressStruct, (const unsigned char*)imageRawData, fileInfo.st_size);
-    jpeg_read_header(&decompressStruct, true);
+    jpeg_read_header(&decompressStruct, (boolean)true);
 
 #if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
-    decompressStruct.out_color_space = JCS_EXT_BGRA;
+    decompressStruct.out_color_space = JCS_RGB;
 #else
-    decompressStruct.out_color_space = JCS_EXT_ARGB;
+    decompressStruct.out_color_space = JCS_RGB;
 #endif
 
     // decompress
